packages/repository/src/repositories/relation.factory.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/example-todo
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {EntityCrudRepository} from '.';
+import {
+  Model,
+  RelationType,
+  AnyObject,
+  Options,
+  Entity,
+  Class,
+  FilterBuilder,
+  Filter,
+  WhereBuilder,
+  Where,
+  DataObject,
+} from '..';
+import {cloneDeep, isArray} from 'lodash';
+import {
+  HasManyEntityCrudRepository,
+  DefaultHasManyEntityCrudRepository,
+} from './relation.repository';
+
+export declare type RelationDefinitionBase = {
```

Comment:
Why `declare`? Please note `declare` is used to write definitions for another existing JS module.

Reply:
Probably because I copied it over from https://github.com/strongloop/loopback-datasource-juggler/blob/master/types/relation.d.ts#L32 😅.